flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/writer/TestSinkInitContext.java

```diff
@@ -48,7 +48,7 @@
 import java.util.concurrent.ScheduledFuture;
 
 /** A mock implementation of a {@code Sink.InitContext} to be used in sink unit tests. */
-public class TestSinkInitContext implements Sink.InitContext {
+public class TestSinkInitContext implements WriterInitContext {
```

Comment:
Hi @pvary, can we revert this change and provide a new class implements `WriterInitContext`?
It may cause compatibility issues of the test.

Reply:
```java
    @Deprecated
    SinkWriter<InputT> createWriter(InitContext context) throws IOException;

    default SinkWriter<InputT> createWriter(WriterInitContext context) throws IOException {
        return createWriter(new InitContextWrapper(context));
    }
```

```java
TestSinkInitContext initContext = new TestSinkInitContext();
SinkWriter writer = sink.createWriter(initContext);
```

One solution is to not directly construct a Writer through TestSinkInitContext, but instead use Sink.createWriter() to create a Writer. 
So there is no need to revert it change. 
Sorry for the disturbance.